metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -314,7 +314,7 @@ final class ReferenceProvider(
     val realName =
       if (isBackticked) name.substring(1, name.length() - 1)
       else name
-    if (symbol.isLocal || symbol.contains(realName)) {
+    if (symbol.isLocal || symbol.desc.name.toString == realName) {
```

Comment:
This is because I figured contains is not good enough here. And this is only used in case of rename, which doesn't make references slower and we are fine with some flaky results there, but we do not want to break the code when renaming.